apps/ejabberd/src/cyrsasl.erl

```diff
@@ -36,15 +36,12 @@
 
 -include("ejabberd.hrl").
 
--record(sasl_mechanism, {mechanism :: mechanism(),
+-record(sasl_mechanism, {mechanism :: mechanism() | atom(), % atom() used in ets:select
                          module :: sasl_module(),
-                         password_type :: plain | digest | scram
+                         password_type :: plain | digest | scram | atom() % atom() used in ets:select
                         }).
--type sasl_module() :: cyrsasl_anonymous
-                     | cyrsasl_digest
-                     | cyrsasl_plain.
+-type sasl_module() :: atom().
```

Comment:
Because the ideal situation would be to have this module completely agnostic of available SASL methods and using `atom()` (OK, maybe `module()` would be a better description) is a first step towards this direction. Take a look - this list lacked `cyrsasl_scram` and some new mech can appear sooner or later.